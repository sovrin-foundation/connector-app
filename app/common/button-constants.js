// @flow
export const DENY = 'Deny'
export const CONNECT = 'Connect'
