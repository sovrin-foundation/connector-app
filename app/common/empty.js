// @flow
// sometimes we need an empty object to use as a placeholder in destructuring
export default {}
