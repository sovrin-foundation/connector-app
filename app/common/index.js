// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
