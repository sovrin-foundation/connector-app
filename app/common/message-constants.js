// @flow
// for overall app message string, it can be error, success, warning etc.

export const TOUCH_ID_MESSAGE = 'Please confirm with Biometrics'
export const TOUCH_ID_NOT_AVAILABLE =
  'Biometrics is not supported on this device'

export const DEVICE_ENROLLMENT_ERROR = [
  'Enrollment Error',
  'Identifier or Seed or Push-notification token not present!',
]

export const PUSH_NOTIFICATION_PERMISSION_ERROR = [
  'Permission Error',
  'Push Notification permission needed! Please enable it from Settings.',
]

export const DUPLICATE_CONNECTION_ERROR = [
  'Attention!',
  'Duplicate Connection invitation!',
]

export const UNLOCKING_APP_WAIT_MESSAGE = 'Loading please wait...'

export const ENTER_PASS_CODE_MESSAGE = 'Enter passcode'

export const ENTER_YOUR_PASS_CODE_MESSAGE = 'Enter your passcode'

export const FIRST_BACKUP_TITLE = 'Set up data recovery now!'

export const FIRST_BACKUP_SUBTEXT = '2 easy steps, takes 3 minutes.'

export const SUBSEQUENT_BACKUP_TITLE =
  'You have new data that is not backed up.'

export const SUBSEQUENT_BACKUP_SUBTEXT =
  "Create another backup so you don't lose anything."

export const PASSPHRASE_GENERATION_ERROR =
  'Uh oh... Looks like something went wrong.'
