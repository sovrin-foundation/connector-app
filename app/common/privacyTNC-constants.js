// @flow

import { Platform } from 'react-native'

const isAndroid = Platform.OS === 'android'
export { EULA_URL as TermsAndConditionUrl } from '../eula/type-eula'
export const TermsAndConditionsTitle = 'Terms and Conditions'

export const PrivacyPolicyUrl = 'https://www.connect.me/privacy.html'
export const localPrivacyPolicySource = isAndroid
  ? 'file:///android_asset/external/connectme/privacy.html'
  : './privacy.html'
export const PrivacyPolicyTitle = 'Privacy Policy'
