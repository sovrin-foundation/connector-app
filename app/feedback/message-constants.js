// @flow
import { Platform } from 'react-native'

export const apptentiveCredentials = Platform.select({
  ios: {
    apptentiveKey: '***REMOVED***',
    apptentiveSignature: '***REMOVED***',
  },
  android: {
    apptentiveKey: '***REMOVED***',
    apptentiveSignature: '***REMOVED***',
  },
})
