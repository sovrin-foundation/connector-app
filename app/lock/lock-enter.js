// @flow
import React, { PureComponent } from 'react'
import {
  InteractionManager,
  StyleSheet,
  Platform,
  Dimensions,
  Image,
} from 'react-native'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import {
  Container,
  CustomText,
  PinCodeBox,
  CustomView,
  Icon,
} from '../components'
import {
  color,
  OFFSET_1X,
  OFFSET_2X,
  OFFSET_3X,
  OFFSET_4X,
} from '../common/styles'
import { CHECK_PIN_IDLE, CHECK_PIN_SUCCESS, CHECK_PIN_FAIL } from './type-lock'
import type {
  LockEnterPinProps,
  LockEnterState,
  CheckPinStatus,
  LockEnterProps,
} from './type-lock'
import { checkPinAction, checkPinStatusIdle } from './lock-store'
import { switchErrorAlerts } from '../store/config-store'
import type { Store } from '../store/type-store'
import { ENTER_YOUR_PASS_CODE_MESSAGE } from '../common/message-constants'
import {
  whiteSmokeRGBA,
  whisper,
  isBiggerThanShortDevice,
  grey,
  matterhornSecondary,
} from '../common/styles/constant'

const lockImage = require('../images/lockCombo.png')
const backgroundImg = require('../images/wave1.png')

const styles = StyleSheet.create({
  //TODO : add animations when keyboard popups.
  text: {
    marginTop: OFFSET_4X,
    minHeight: OFFSET_3X,
    marginBottom: OFFSET_4X,
  },
  titleText: {
    lineHeight: 28,
    letterSpacing: 0.5,
  },
})

export const WrongPinText = (
  <CustomText h4 bg="tertiary" tertiary center>
    Wrong passcode! Please try again
  </CustomText>
)

export class LockEnter extends PureComponent<LockEnterProps, LockEnterState> {
  pinCodeBox = null
  clearFailStatus = () => {
    this.props.checkPinStatusIdle()
  }

  clearFailStatusDelayed = () => {
    setTimeout(this.clearFailStatus, 1000)
  }

  onPinComplete = (pin: string) => {
    this.props.checkPinAction(pin)
  }

  componentWillReceiveProps(nextProps: LockEnterProps) {
    if (this.props.checkPinStatus !== nextProps.checkPinStatus) {
      if (nextProps.checkPinStatus === CHECK_PIN_SUCCESS) {
        this.pinCodeBox && this.pinCodeBox.hideKeyboard()
        this.props.onSuccess()
      } else if (nextProps.checkPinStatus === CHECK_PIN_FAIL) {
        this.pinCodeBox && this.pinCodeBox.clear && this.pinCodeBox.clear()
        // set status back to idle so we can come to this else again
        this.clearFailStatusDelayed()
      }
    }
  }

  componentDidMount() {
    if (this.props.checkPinStatus === CHECK_PIN_SUCCESS) {
      this.clearFailStatus()
    }
  }

  render() {
    const {
      checkPinStatus,
      message = ENTER_YOUR_PASS_CODE_MESSAGE,
    } = this.props

    const EnterPinText = (
      <CustomText
        style={[styles.titleText]}
        center
        h4
        bg="tertiary"
        tertiary
        thick
        testID="pass-code-input-text"
      >
        {message}
      </CustomText>
    )

    return (
      <Container>
        {this.props.fromRecovery ? (
          <Container safeArea fifth>
            <Image
              source={backgroundImg}
              style={[stylesRecovery.backgroundImg]}
            />
            <CustomView
              center
              transparentBg
              style={[stylesRecovery.topLockIcon]}
            >
              <CustomView style={[stylesRecovery.blackStrip]} />
              <CustomView style={[stylesRecovery.lockIconWrapper]}>
                <Icon extraLarge halo src={lockImage} />
              </CustomView>
            </CustomView>
            <CustomView center verticalSpace>
              <CustomText
                center
                transparentBg
                style={[stylesRecovery.lockHeading]}
                heavy
                charcoal
              >
                Please enter your current Connect.Me passcode!
              </CustomText>
            </CustomView>
            <CustomView center>
              <PinCodeBox
                ref={pinCodeBox => {
                  this.pinCodeBox = pinCodeBox
                }}
                onPinComplete={this.onPinComplete}
              />
            </CustomView>
            <CustomView center doubleVerticalSpace>
              <CustomText
                transparentBg
                center
                h6
                bold
                style={[stylesRecovery.newPasscodeText]}
                onPress={this.props.setupNewPassCode}
                testID={'set-up-new-passcode-recovery'}
              >
                Or Setup New Passcode
              </CustomText>
            </CustomView>
          </Container>
        ) : (
          <Container tertiary>
            <CustomView
              style={[styles.text]}
              center
              onPress={this.props.switchErrorAlerts}
              testID="pin-code-input-boxes"
            >
              {checkPinStatus === CHECK_PIN_IDLE && EnterPinText}
              {checkPinStatus === CHECK_PIN_SUCCESS && EnterPinText}
              {checkPinStatus === CHECK_PIN_FAIL && WrongPinText}
            </CustomView>
            <CustomView center>
              <PinCodeBox
                ref={pinCodeBox => {
                  this.pinCodeBox = pinCodeBox
                }}
                onPinComplete={this.onPinComplete}
                enableCustomKeyboard={this.props.enableCustomKeyboard}
              />
            </CustomView>
          </Container>
        )}
      </Container>
    )
  }
}

const stylesRecovery = StyleSheet.create({
  backgroundImg: {
    position: 'absolute',
    transform: [{ rotate: '-180deg' }],
    marginTop: '-92%',
    width: '100%',
  },
  topLockIcon: {
    ...Platform.select({
      ios: {
        marginTop: isBiggerThanShortDevice ? '14%' : '3%',
      },
      android: {
        marginTop: isBiggerThanShortDevice ? '12%' : '8%',
      },
    }),
  },
  blackStrip: {
    position: 'absolute',
    height: 8,
    backgroundColor: matterhornSecondary,
    width: '100%',
  },
  lockIconWrapper: {
    backgroundColor: matterhornSecondary,
    borderRadius: 50,
  },
  lockHeading: {
    fontSize: isBiggerThanShortDevice ? 23 : 16,
    width: '80%',
    ...Platform.select({
      ios: {
        marginTop: isBiggerThanShortDevice ? '7%' : '2%',
        marginBottom: '-1%',
      },
      android: {
        marginTop: isBiggerThanShortDevice ? '2%' : 0,
      },
    }),
  },
  newPasscodeText: {
    ...Platform.select({
      ios: {
        marginTop: isBiggerThanShortDevice ? '6.1%' : '1%',
      },
      android: {
        marginTop: isBiggerThanShortDevice ? '2%' : 0,
      },
    }),
    color: color.bg.fifteenth.color,
  },
})

const mapStateToProps = (state: Store) => ({
  checkPinStatus: state.lock.checkPinStatus,
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      checkPinAction,
      checkPinStatusIdle,
      switchErrorAlerts,
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(LockEnter)
