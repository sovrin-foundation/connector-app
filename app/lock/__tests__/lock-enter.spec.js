// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { LockEnter } from '../lock-enter'
import { CHECK_PIN_IDLE, CHECK_PIN_SUCCESS, CHECK_PIN_FAIL } from '../type-lock'
import { ENTER_YOUR_PASS_CODE_MESSAGE } from '../../common/message-constants'

describe('<LockEnter />', () => {
  function getProps() {
    return {
      checkPinAction: jest.fn(),
      checkPinStatusIdle: jest.fn(),
      switchErrorAlerts: jest.fn(),
      onSuccess: jest.fn(),
      message: ENTER_YOUR_PASS_CODE_MESSAGE,
      checkPinStatus: CHECK_PIN_IDLE,
    }
  }

  let component
  let props
  let cleared

  const options = {
    createNodeMock: element => {
      return {
        clear: () => {
          cleared = true
        },
        blur: () => {},
      }
    },
  }

  beforeEach(() => {
    props = getProps()
    component = renderer.create(
      <LockEnter {...props} fromRecovery={false} />,
      options
    )
  })

  it('should match snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should match snapshot for recovery flow', () => {
    const component1 = renderer.create(
      <LockEnter {...props} fromRecovery={true} />,
      options
    )
    expect(component1.toJSON()).toMatchSnapshot()
  })

  it('call onSuccess if pin matches', () => {
    component.update(
      <LockEnter
        {...props}
        checkPinStatus={CHECK_PIN_SUCCESS}
        fromRecovery={false}
      />
    )
    expect(props.onSuccess).toHaveBeenCalled()
  })

  it('clear pin code box if pin check fails', () => {
    jest.useFakeTimers()
    component.update(
      <LockEnter
        {...props}
        checkPinStatus={CHECK_PIN_FAIL}
        fromRecovery={false}
      />
    )
    jest.runAllTimers()
    expect(props.checkPinStatusIdle).toHaveBeenCalled()
  })
})
