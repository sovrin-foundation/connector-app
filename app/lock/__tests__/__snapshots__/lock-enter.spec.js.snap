// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LockEnter /> should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }
    tertiary={true}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="pin-code-input-boxes-touchable"
      accessibilityTraits={undefined}
      accessible={false}
      center={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 40,
          "marginTop": 40,
          "minHeight": 30,
        }
      }
      testID="pin-code-input-boxes-touchable"
    >
      <Text
        accessibilityLabel="pass-code-input-text"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "800",
            },
            Object {
              "textAlign": "center",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "letterSpacing": 0.5,
              "lineHeight": 28,
            },
          ]
        }
        testID="pass-code-input-text"
      >
        Enter your passcode
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={
            Object {
              "bottom": 20,
              "left": 20,
              "right": 20,
              "top": 20,
            }
          }
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          row={true}
          style={
            Object {
              "flexDirection": "row",
            }
          }
          testID={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-0-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-0-touchable"
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-1-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-1-touchable"
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-2-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-2-touchable"
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-3-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-3-touchable"
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-4-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-4-touchable"
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-5-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-5-touchable"
          />
        </View>
        <TextInput
          accessibilityLabel="pin-code-input-box"
          accessible={true}
          autoCorrect={false}
          autoFocus={true}
          blurOnSubmit={false}
          enablesReturnKeyAutomatically={false}
          keyboardAppearance="dark"
          keyboardType="number-pad"
          maxLength={6}
          onChangeText={[Function]}
          style={
            Object {
              "height": 0,
              "position": "absolute",
              "right": -999,
              "width": 0,
            }
          }
          testID="pin-code-input-box"
          value=""
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`<LockEnter /> should match snapshot for recovery flow 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    fifth={true}
    safeArea={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "marginTop": "-92%",
            "position": "absolute",
            "transform": Array [
              Object {
                "rotate": "-180deg",
              },
            ],
            "width": "100%",
          },
        ]
      }
    />
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": "3%",
        }
      }
      transparentBg={true}
    >
      <View
        style={
          Object {
            "backgroundColor": "#505050",
            "height": 8,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#505050",
            "borderRadius": 50,
          }
        }
      >
        <View
          clearBg={true}
          extraLarge={true}
          halo={true}
          src={1}
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <View
            center={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0,0,0,0.07)",
                "borderRadius": 50,
                "height": 100,
                "justifyContent": "center",
                "width": 100,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Image
                resizeMode="contain"
                source={1}
                style={
                  Array [
                    Object {
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "borderRadius": 40,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingVertical": 8,
        }
      }
      verticalSpace={true}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            Object {
              "fontWeight": "900",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "#464646",
            },
            null,
            null,
            Object {
              "fontSize": 16,
              "marginBottom": "-1%",
              "marginTop": "2%",
              "width": "80%",
            },
          ]
        }
      >
        Please enter your current Connect.Me passcode!
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={
            Object {
              "bottom": 20,
              "left": 20,
              "right": 20,
              "top": 20,
            }
          }
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          row={true}
          style={
            Object {
              "flexDirection": "row",
            }
          }
          testID={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-0-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-0-touchable"
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-1-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-1-touchable"
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-2-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-2-touchable"
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-3-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-3-touchable"
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-4-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-4-touchable"
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="pin-code-digit-5-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            center={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#777777",
                "borderBottomWidth": 5,
                "height": 66,
                "justifyContent": "center",
                "marginHorizontal": 5,
                "width": 38,
              }
            }
            testID="pin-code-digit-5-touchable"
          />
        </View>
        <TextInput
          accessibilityLabel="pin-code-input-box"
          accessible={true}
          autoCorrect={false}
          autoFocus={true}
          blurOnSubmit={false}
          enablesReturnKeyAutomatically={false}
          keyboardAppearance="dark"
          keyboardType="number-pad"
          maxLength={6}
          onChangeText={[Function]}
          style={
            Object {
              "height": 0,
              "position": "absolute",
              "right": -999,
              "width": 0,
            }
          }
          testID="pin-code-input-box"
          value=""
        />
      </View>
    </View>
    <View
      center={true}
      doubleVerticalSpace={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingVertical": 16,
        }
      }
    >
      <Text
        accessibilityLabel="set-up-new-passcode-recovery"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "#79ade7",
              "marginTop": "1%",
            },
          ]
        }
        testID="set-up-new-passcode-recovery"
      >
        Or Setup New Passcode
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;
